_impls/thrussh.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Thrussh
+homepage: http://pijul.org/thrussh
+source-repository: https://nest.pijul.com/pijul_org/thrussh
+license: "[Apache-2.0](https://www.apache.org/licenses/LICENSE-2.0)"
+first-release:
+    date: 2016-07-01
+latest-release:
+    version: 0.10.3
+    date: 2017-05-07
+client: yes
+server: yes
+
+protocols:
+    cipher:
+        - chacha20-poly1305@openssh.com
+    compression:
+        - none
+    hostkey:
+        - rsa-sha2-256
+        - rsa-sha2-512
+        - ssh-ed25519                       # since 6.5
+    kex:
+        - curve25519-sha256@libssh.org      # since 6.5
+        - ecdh-sha2-nistp256                # since 5.7
+    mac:
```

Comment:
That's right, Thrussh doesn't yet support MACs, because it only supports one cipher in its current version, and that cipher includes MAC already.

It is actually a feature of modern crypto libraries (such as [\*ring\*](https://github.com/briansmith/ring), upon which Thrussh is based), to not even include separate ciphers and MACs.

Thrussh might support MAC in the future though, if there is a need for it.